validator/sawtooth_validator/execution/scheduler_parallel.py

```diff
@@ -77,6 +77,18 @@ def _tokenize_address(self, address):
             for i in range(0, len(address), self._token_size)
         ]
 
+    def _get_child_and_remainder(self, node, address):
+        try:
+            child, addr = next(
+                (child, addr)
+                for addr, child in node.children.items()
+                if address.startswith(addr[:self._token_size])
+            )
```

Comment:
Why is this implementation preferred to the following?
```
for addr, child in node.children.items() :
    if address.startswith(addr[:self._token_size])
        return child, addr
raise AddressNotInTree()
```

This implementation seems harder to follow and less standard.